settings/R/MultiSettings.R

```diff
@@ -0,0 +1,195 @@
+      return(result[[1]])
+    } else if(is.list(result[[1]]) && all(sapply(result[[1]], is.Settings))) {
+      result <- result[[1]]
+    }
+  }
+  
+  if(!all(sapply(result, is.Settings))) {
+    stop("MultiSettings can only be made from Setting, MultiSettings, or a list of Settings")
+  }
+
+  if(length(result) > 0)
+    names(result) <- paste("settings", seq_along(result), sep=".")
+  class(result) <- c("MultiSettings", class(result))
+  return(result)
+}
+
+
+##' @export
+##' @describeIn 
+as.MultiSettings <- function(x) {
+  return(MultiSettings(x))
+}
+
+##' @export
+is.MultiSettings <- function(x) {
+  return(is(x, "MultiSettings"))
+}
+
+##' @export
+"[[<-.MultiSettings" <- function(x, value, i, global=TRUE) {
+  if(is.character(i)) {
+    if(global) {
+      value <- replicate(length(x), value, simplify=FALSE)
+      x[[i, global=F]] <- value
+    } else {
+      if(length(x) == length(value)) {
+        value <- as.list(value)
+        if(is.null(names(value))) {
+          names(value) <- rep(i, length(x))
+        }
+        for(j in seq_along(x)) {
+          x[[j]][[i]] <- value[[j]]
+        }
+      } else if(length(x) == 1 && length(value) > 1) {
+        x <- MultiSettings(replicate(length(value), x[[1]], simplify=FALSE))
+        x[[i, global=FALSE]] <- value
+      } else {
+        stop("Length mismatch in assigning to MultiSettings")
+      }
+    }
+    return(x)
+  } else {
+    if(is.Settings(value) || is.null(value)) {
+      NextMethod()
+    } else {
+      stop("Can only add a Settings object to MultiSettings.")
+    }
+  }
+}
+
+##' @export
+"$<-.MultiSettings" <- function(x, value, i, global=TRUE) {
+  return("[[<-.MultiSettings"(x, value, i, global))
+}
+
+
+##' @export
+"[<-.MultiSettings" <- function(x, value, i) {
+  stop("MultiSettings don't support assignments using '['")
+}
+
+
+##' @export
+"[[.MultiSettings" <- function(x, i, collapse=TRUE, setAttributes=FALSE) {
+  if(is.character(i)) {
+    result <- lapply(x, function(y) y[[i]])
+    if(collapse && .allListElementsEqual(result)) {
+      result <- result[[1]]
+      if(setAttributes) {
+        attr(result, "settingType") <- "global"
+      }
+    } else {
+      if(setAttributes) {
+        attr(result, "settingType") <- "multi"
+      }
+    }
+    return(result)
+  } else {
+    NextMethod()
+  }
+}
+
+.allListElementsEqual <- function(x) {
+  firstElement <- x[[1]]
+  replicatedFirstElement <- replicate(length(x), firstElement, simplify=FALSE)
+  return(isTRUE(all.equal(replicatedFirstElement, x, check.attributes=FALSE)))
+}
+
+##' @export
+"$.MultiSettings" <- function(x, i) {
+  return(x[[i]])
+}
+
+
+##' @export
+"[.MultiSettings" <- function(x, i) {
+  if(is.character(i)) {
+    stop("MultiSettings don't support selecting by '[' with character indices")
+  } else {
+    return(MultiSettings(NextMethod()))
+  }
+}
+
+##' @export
+names.MultiSettings <- function(x) {
+  return(unique(unlist(lapply(x, names))))
```

Comment:
> More generally, all of these functions need better documentation. At a minimum, they need more roxygen tags other than just export

I know, just wanted to get this up before leaving for vacation so you can see what I'm up to. 

To answer your question, the vector of names returned here will be the names of the elements of a single Settings in the MultiSettings. I.e., it's just supposed to give the same thing as `names(settings)` would for a normal Settings object. And it will, except in the (unusual, I would think) case that different Settings in the list have different elements. In that case I didn't want elements to be lost entirely, so I thought listing the union of all names was better than nothing. This is meant to be mostly a convenience function, but at the same time work as if MultiSettings was just a regular Settings in typical use cases.